ddtrace/tracer/log.go

```diff
@@ -107,6 +109,13 @@ func logStartup(t *tracer) {
 	if _, err := samplingRulesFromEnv(); err != nil {
 		info.SamplingRulesError = fmt.Sprintf("%s", err)
 	}
+	// Based on the apply logic in sampler.go func (*rulesSampler).applyRate,
```

Comment:
I generally tried to apply the coding style of putting inline comments referring to `if` statements inside the `if` statement. I think I added this into the coding style too [somewhere](https://github.com/DataDog/dd-trace-go/wiki/Style-guidelines#comments). The reasoning being that if you're not interested in reading down that code path, then the comment is not important either. Basically the comment only applies once the condition is true.

I'd also abstain from calling out unexported methods. The chance of refactoring those or renaming them and forgetting to update this comment is high. I'd rather explain here why this logic is like it is, rather than say "this is so because of that other logic". One day that other logic will change.